### ISSUE TITLE
More prebuilt indexes features

### ISSUE THREAD

**Issue by lintool** [castorini/pyserini#233]
hi @qguo96 - I just tried the new prebuilt indexes feature... it's awesome!

```python
>>> from pyserini.search import SimpleSearcher
>>> searcher = SimpleSearcher.from_prebuilt_index('trec45')
Downloading index at https://git.uwaterloo.ca/jimmylin/anserini-indexes/raw/master/index-robust04-20191213.tar.gz...
index-robust04-20191213.tar.gz: 1.70GB [00:50, 36.4MB/s]                                                                                                                                                                        
Extracting /Users/jimmylin/.cache/pyserini/indexes/index-robust04-20191213.tar.gz into /Users/jimmylin/.cache/pyserini/indexes/index-robust04-2019121315f3d001489c97849a010b0a4734d018...
>>> searcher
<pyserini.search._searcher.SimpleSearcher object at 0x7fee58547ac8>
>>> hits = searcher.search('hubble space telescope')
>>> 
>>> # Print the first 10 hits:
... for i in range(0, 10):
...     print(f'{i+1:2} {hits[i].docid:15} {hits[i].score:.5f}')
... 
 1 LA071090-0047   16.85690
 2 FT934-5418      16.75630
 3 FT921-7107      16.68290
 4 LA052890-0021   16.37390
 5 LA070990-0052   16.36460
 6 LA062990-0180   16.19260
 7 LA070890-0154   16.15610
 8 FT934-2516      16.08950
 9 LA041090-0148   16.08810
10 FT944-128       16.01920
```

A minor request:

```
Extracting /Users/jimmylin/.cache/pyserini/indexes/index-robust04-20191213.tar.gz into /Users/jimmylin/.cache/pyserini/indexes/index-robust04-2019121315f3d001489c97849a010b0a4734d018...
```

Can we add a `.` between the end of the index name and the MD5 checksum?

In addition, it'd be nice to have a feature like `SimpleSearcher.list_prebuilt_indexes()` where it shows information about available prebuilt indexes, maybe as a DF? The columns of the DF might be something like:

+ name
+ short string description
+ download URL
+ downloaded (e.g., cached)
+ size compressed
+ size uncompressed
+ etc.

Thoughts?

**Comment by lintool** [comment]
Right now, we can do this:

```python
>>> from pyserini.index import IndexReader
>>> index_reader = IndexReader('indexes/index-robust04-20191213/')
>>> index_reader.stats()
{'total_terms': 174540872, 'documents': 528030, 'non_empty_documents': 528030, 'unique_terms': 923436}
```

(1) the stats should probably also go into the DF referenced above.
(2) `IndexReader.from_prebuilt_index('trec45')` should work the same way?

**Comment by qguo96** [comment]
Sure, I will add these features later!

**Comment by qguo96** [comment]
Hi @lintool, I'm considering to use a CSV file to store the information of indexes. Is this a good idea?

**Comment by lintool** [comment]
nah, I think hard-coded JSON structures in code is fine for now...

**Comment by lintool** [comment]
Closed by #235
